src/block_cache/block_mgr.c

```diff
@@ -494,22 +557,55 @@ __bm_stat(WT_BM *bm, WT_SESSION_IMPL *session, WT_DSRC_STATS *stats)
 
 /*
  * __bm_switch_object --
- *     Modify the tiered object.
+ *     Switch the tiered object.
  */
 static int
-__bm_switch_object(WT_BM *bm, WT_SESSION_IMPL *session, uint32_t object_id, uint32_t flags)
+__bm_switch_object(WT_BM *bm, WT_SESSION_IMPL *session, uint32_t objectid, uint32_t flags)
 {
-    return (__wt_block_switch_object(session, bm->block, object_id, flags));
+    WT_BLOCK *block;
+
+    block = bm->block;
+
+    /* Close out our current handle. */
+    WT_RET(__bm_close_block(session, block));
+    bm->block = NULL;
+
+    /*
+     * FIXME-WT-7596 the flags argument will be used in the future to perform various tasks,
+     * to efficiently mark objects in transition (that is during a switch):
+     *  - mark this file as the writeable file (what currently happens)
+     *  - disallow writes to this object (reads still allowed, we're about to switch)
+     *  - close this object (about to move it, don't allow reopens yet)
+     *  - allow opens on this object again
+     */
+    WT_UNUSED(flags);
+    WT_RET(__wt_blkcache_get_handle(session, NULL, objectid, &block));
+
+    /*
+     * KEITH XXX: We need to distinguish between tiered switch and loading a checkpoint. This is
+     * also discarding the extent list which isn't correct, because we can't know when to discard
+     * previous files if we don't have the extent list. This fixes the problem where we randomly
```

Comment:
Is this something we are comfortable with as future work if we accept this change?  Or is this an unknown that we need to figure out how to address before accepting this change?

Reply:
@ddanderson should comment, but I believe this comment simply notes the current state of the tiered storage project: tiered storage can't create "valid" database objects yet, and switching between writeable files is part of that.